public/js/post/post.js

```diff
@@ -63,47 +63,59 @@ var MultipleCourseCode = React.createClass({
             infoOpened: false
         }
     },
+
+    componentDidMount: function() {
+        for (i = 0; i < this.props.data.courses.length; i++) {
+            this.setState({completedTextBoxes: this.state.completedTextBoxes += 1});
```

Comment:
@christinem Ah. That's because `setState` is asynchronous. Try:
```JavaScript
this.setState({
    completedTextBoxes: this.state.completedTextBoxes + this.props.data.courses.length
}, this.checkIfCompleted);
```